### PR TITLE
Format docs directory with black

### PR DESCRIPTION
**Pull request by klecki**
## Description:
This PR applies Black to the docs/ directory and enables linter on this path.

The `examples/use_cases` subdirectory is excluded - it contains a lot of 3rd party
code that was forked by DALI and has numerous linter issues. 
Linting and formatting that part should be done as a separate step (if at all).

 #5226 enabled black to be used within jupyter files on our CI.

After adjusting the configs, the `docs/` directory is formatted with black.
Next, all linter issues are resolved.

Notable changes:
1. The index file generation for operator references gets a header with imports and a license.
This solves a lot of `F821 undefined name XXXX`. Please check that the references are correctly
generated in docs.
If advised, I can extract this change to a separate PR.

2. Some too-long strings and too-long comments are broken into additional lines
3. Several one-letter variables are replaced with more meaningful ones 
4. noqa is added around mxnet imports that appear to be no-op, but it looks like the importing has side effects (for example path adjustment to look up the library).
5. Some fixes in the conf.py - please verify again that the documentation is still rendered correctly.

### Affected modules and functionalities:
Docs, jupyter examples.

### Key points relevant for the review:
Check if example references are correct, the enums are rendered correctly in the docs, versioning and the drop down for stable/nightly docs selector work correctly.

Do we need to split this into:
* config
* separate formatting step that will be excluded from the blame

Most of the formatting changes touch jupyter which is typically not viewed via blame-aware tools.

### Tests:

- [x] Existing tests apply
 The jupyters should run correctly.
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [x] Documentation updated
Examples were reformatted.
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [ ] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by klecki** [comment]
!build

File: docs/examples/frameworks/mxnet/demo/common/fit.py [hunk 1]
```diff
@@ -70,71 +67,95 @@ def _save_model(args, rank=0):
...
+    train.add_argument(
+        "--top-k", type=int, default=0, help="report the top-k accuracy. 0 means no report."
+    )
+    train.add_argument(
+        "--loss",
+        type=str,
+        default="",
+        help=(
+            "show the cross-entropy or nll loss."
+            " ce strands for cross-entropy, nll-loss stands for likelihood loss"
```
Comment (stiepan) [inline docs/examples/frameworks/mxnet/demo/common/fit.py:126]: Is it a PR where we fix misspelled words, likely not, but:

```suggestion
            " ce stands for cross-entropy, nll-loss stands for likelihood loss"
```

**Review by stiepan — APPROVED** [review]

File: docs/examples/frameworks/mxnet/demo/common/fit.py (hunk 1 above)
Comment (klecki) [reply docs/examples/frameworks/mxnet/demo/common/fit.py]: done

**Review by awolant — APPROVED** [review]

**Comment by klecki** [comment]
!build

**Review by szalpal — APPROVED** [review]

**Pull request merged**